nav2_map_server/include/nav2_map_server/occ_grid_loader.hpp

```diff
@@ -12,65 +12,69 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#ifndef NAV2_MAP_SERVER__MAP_REPS__OCC_GRID_SERVER_HPP_
-#define NAV2_MAP_SERVER__MAP_REPS__OCC_GRID_SERVER_HPP_
+#ifndef NAV2_MAP_SERVER__OCC_GRID_LOADER_HPP_
+#define NAV2_MAP_SERVER__OCC_GRID_LOADER_HPP_
 
-#include <memory>
 #include <string>
+#include <vector>
 
+#include "rclcpp/rclcpp.hpp"
 #include "nav_msgs/msg/occupancy_grid.hpp"
 #include "nav_msgs/msg/map_meta_data.hpp"
 #include "nav_msgs/srv/get_map.hpp"
-#include "rclcpp/rclcpp.hpp"
-#include "nav2_map_server/base_map_server.hpp"
+#include "nav2_map_server/map_loader.hpp"
 
 namespace nav2_map_server
 {
 
-class OccGridServer : public BaseMapServer
+class OccGridLoader : public MapLoader
 {
 public:
-  explicit OccGridServer(rclcpp::Node::SharedPtr node)
-  : node_(node) {}
-  OccGridServer(rclcpp::Node::SharedPtr node, std::string file_name);
-  OccGridServer() {}
-  ~OccGridServer() {}
-
-  void LoadMapInfoFromFile(const std::string & file_name);
-  void LoadMapFromFile(const std::string & map_name);
-  void PublishMap();
-  void SetMap();
-  void ConnectROS();
+  explicit OccGridLoader(rclcpp::Node * node);
+  OccGridLoader() = delete;
+
+  // Load the image and generate an occupancy grid
+  void loadMapFromFile(const std::string & filename) override;
+
+  // Make the occupancy griad available via ROS topic and service
+  void initServices() override;
 
 protected:
-  enum MapMode { TRINARY, SCALE, RAW };
+  // The frame ID used in the returned occupancy grid message
+  static const char * frame_id_;
 
-  // Info from parsing the YAML file
-  double origin_[3];
+  // The name for the topic on which the map will be published
+  static const char * topic_name_;
+
+  // The name of the service to GetMap
+  static const char * service_name_;
+
+  // Map parameters retrieved from the node
+  double resolution_;
   int negate_;
-  double occ_th_;
-  double free_th_;
-  double res_;
+  double occupied_thresh_;
+  double free_thresh_;
+  enum MapMode { TRINARY, SCALE, RAW };
   MapMode mode_ = TRINARY;
-  std::string frame_id_ = "map";
-  std::string map_name_;
+  std::vector<double> origin_;
 
   // The ROS node to use for ROS-related operations such as creating a service
-  rclcpp::Node::SharedPtr node_;
+  rclcpp::Node * node_;
```

Comment:
Shouldn't this be a boost shared pointer to fall in line with other node's convention?